custom/icds/location_reassignment/utils.py

```diff
@@ -52,21 +52,26 @@ def get_household_and_child_case_ids_by_owner(domain, household_case_id, owner_i
 
 
 def get_household_child_cases_by_owner(domain, household_case_id, owner_id, case_types):
-    def get_child_cases(ids):
+    def get_child_cases(ids, exclude_ids):
         return [case for case in
                 CaseAccessorSQL.get_reverse_indexed_cases(domain, ids)
-                if case.owner_id == owner_id]
+                if case.case_id not in exclude_ids and case.owner_id == owner_id]
 
+    # keep a list of case ids already iterated to avoid duplicates
+    iterated_case_ids = set()
     cases = set()
     parent_case_ids = [household_case_id]
     while parent_case_ids:
-        child_cases = get_child_cases(parent_case_ids)
+        child_cases = [case for case in get_child_cases(parent_case_ids, iterated_case_ids)]
```

Comment:
updated [later](https://github.com/dimagi/commcare-hq/pull/27153/commits/e81fcda4d951aef3e8b03051937086b15e0575fa) to avoid the redundant iteration.